pkgs/development/python-modules/betterproto/default.nix

```
@@ -0,0 +1,66 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+, pythonOlder
+, poetry-core
+, grpclib
+, python-dateutil
+, dataclasses
+, black
+, jinja2
+, isort
+, callPackage
+}:
+let
+  version = "2.0.0b5";
+in
+buildPythonPackage {
+  pname = "betterproto";
+  inherit version;
+  format = "pyproject";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "danielgtaylor";
+    repo = "python-betterproto";
+    rev = "v${version}";
+    sha256 = "sha256-XyXdpo3Yo4aO1favMWC7i9utz4fNDbKbsnYXJW0b7Gc=";
+  };
+
+  nativeBuildInputs = [ poetry-core ];
+
+  propagatedBuildInputs = [
+    grpclib
+    python-dateutil
+  ] ++ lib.optional (pythonOlder "3.7") [
+    dataclasses
+  ];
+
+  passthru.optional-dependencies.compiler = [
+    black
+    jinja2
+    isort
```

```suggestion
    jinja2
```
Those are only linters.

---

Theses are formatters that betterproto uses to format the generated python code. They are required. Tests fail without them.

---

Do all tests require them? If only a few we should rather disable those tests since black can change formatting with updates which should not break other packages.

---

Compiling anything requires the optional dependecies. Without them only half of the package works and incidentally only half the tests. They are really less "optional" and more of a way to separate compiler from runtime library.

---

Sigh, I hope that black updates to no break it.